File: src/main/java/com/vaadin/demo/component/checkbox/CheckboxHorizontal.java
package com.vaadin.demo.component.checkbox;

import com.vaadin.flow.component.checkbox.CheckboxGroup;
import com.vaadin.flow.component.checkbox.CheckboxGroupVariant;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.router.Route;
import com.vaadin.demo.DemoExporter; // hidden-source-line

@Route("checkbox-horizontal")
public class CheckboxHorizontal extends Div {

    public CheckboxHorizontal() {
        // tag::snippet[]
        CheckboxGroup<String> checkboxGroup = new CheckboxGroup<>();
        checkboxGroup.setLabel("Permissions");
        checkboxGroup.setItems("Read", "Edit", "Delete");
        add(checkboxGroup);
        // end::snippet[]
    }

    public static class Exporter extends DemoExporter<CheckboxHorizontal> { // hidden-source-line
    } // hidden-source-line
}

File: frontend/demo/component/checkbox/checkbox-horizontal.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/checkbox';
import '@vaadin/checkbox-group';
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/demo/theme';

@customElement('checkbox-horizontal')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-checkbox-group label="Permissions">
        <vaadin-checkbox value="read" label="Read"></vaadin-checkbox>
        <vaadin-checkbox value="edit" label="Edit"></vaadin-checkbox>
        <vaadin-checkbox value="delete" label="Delete"></vaadin-checkbox>
      </vaadin-checkbox-group>
      <!-- end::snippet[] -->
    `;
  }
}
